peekingduck/pipeline/nodes/model/jde_mot/jde_files/utils/utils.py

```diff
@@ -0,0 +1,256 @@
+# Modifications copyright 2021 AI Singapore
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#      https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Original copyright (c) 2019 ZhongdaoWang
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+Utilities functions
+"""
+
+from typing import Any, List, Tuple, Union
+import cv2
+import numpy as np
+import torch
+from torchvision.ops import nms
+
+# pylint: disable=invalid-name, no-member
+
+
+def letterbox(
+    img: np.ndarray,
+    height: int = 608,
+    width: int = 1088,
+    color: Tuple[float, float, float] = (127.5, 127.5, 127.5),
+) -> Tuple[np.ndarray, float, float, float]:
+    """Resize a rectangular image to a padded rectangular.
+
+    Args:
+        img (np.ndarray): Image frame
+        height (int, optional): Height of padded image. Defaults to 608.
+        width (int, optional): Width of padded image. Defaults to 1088.
+        color (Tuple[float, float, float], optional): Colour of border
+            around image. Defaults to (127.5, 127.5, 127.5).
+
+    Returns:
+        Tuple[np.ndarray, float, float, float]: Padded rectangular image.
+    """
+    shape = img.shape[:2]  # shape = [height, width]
+    ratio = min(float(height) / shape[0], float(width) / shape[1])
+    new_shape = (
+        round(shape[1] * ratio),
+        round(shape[0] * ratio),
+    )  # new_shape = [width, height]
```

Comment:
avoid inline comment if it causes code to wrap

Reply:
Removed